crates/mem-pool/src/pool.rs

```diff
@@ -445,11 +445,53 @@ impl MemPool {
         mem_block.repackage(withdrawals_count, deposits_count, txs_count)
     }
 
-    /// Reset
+    /// Reset pool
+    ///
+    /// For Full node and Test node:
     /// this method reset the current state of the mem pool
     /// discarded txs & withdrawals will be reinject to pool
+    ///
+    /// For ReadOnly node:
+    /// This function only update the current tip.
+    /// The state reset of readonly only happend when receives states from publisher, see `refresh_mem_block`
     #[instrument(skip_all, fields(old_tip = old_tip.map(|h| display(h.pack())), new_tip = new_tip.map(|h| display(h.pack()))))]
     async fn reset(&mut self, old_tip: Option<H256>, new_tip: Option<H256>) -> Result<()> {
+        match self.node_mode {
+            NodeMode::FullNode | NodeMode::Test => self.reset_full(old_tip, new_tip).await,
+            NodeMode::ReadOnly => self.reset_read_only(old_tip, new_tip).await,
+        }
+    }
+
+    /// Only **ReadOnly** node.
+    /// update current tip.
+    #[instrument(skip_all)]
+    async fn reset_read_only(
+        &mut self,
+        _old_tip: Option<H256>,
+        new_tip: Option<H256>,
+    ) -> Result<()> {
+        let new_tip = match new_tip {
+            Some(block_hash) => block_hash,
+            None => {
+                log::debug!("reset new tip to last valid tip block");
+                self.store.get_last_valid_tip_block_hash()?
+            }
+        };
+        let new_tip_block = self.store.get_block(&new_tip)?.expect("new tip block");
+        self.current_tip = (new_tip, new_tip_block.raw().number().unpack());
+
+        // Publish new tip.
+        let _ = self.new_tip_publisher.send(self.current_tip);
+        if let Some(ref publish) = self.mem_pool_publish_service {
+            publish.new_tip(self.current_tip).await;
```

Comment:
This is only necessary on full node for now. Because we do not support cascading sync (read only node syncing from another read only node) yet.